src/main/java/org/opensearch/ad/settings/AnomalyDetectorSettings.java

```diff
@@ -100,14 +100,18 @@ private AnomalyDetectorSettings() {}
             Setting.Property.Dynamic
         );
 
-    public static final Setting<Long> AD_RESULT_HISTORY_MAX_DOCS = Setting
+    // Opensearch-only setting. Doesn't plan to use the value of the legacy setting
+    // AD_RESULT_HISTORY_MAX_DOCS as that's too low. If the master node uses opendistro code,
+    // it uses the legacy setting. If the master node uses opensearch code, it uses the new setting.
+    public static final Setting<Long> AD_RESULT_HISTORY_MAX_DOCS_PER_SHARD = Setting
         .longSetting(
-            "plugins.anomaly_detection.ad_result_history_max_docs",
-            // Total documents in primary replica.
-            // A single feature result is roughly 150 bytes. Suppose a doc is
-            // of 200 bytes, 250 million docs is of 50 GB. We choose 50 GB
-            // because we have 1 shard at least. One shard can have at most 50 GB.
-            LegacyOpenDistroAnomalyDetectorSettings.AD_RESULT_HISTORY_MAX_DOCS,
+            "plugins.anomaly_detection.ad_result_history_max_docs_per_shard",
+            // Total documents in the primary shards.
+            // Note the count is for Lucene docs. Lucene considers a nested
```

Comment:
This number is used in `CreateIndexRequest#addMaxIndexDocsCondition` for adding condition to check if the index has at least numDocs. Does it mean the lucene doc count or ES index doc count?

Reply:
Lucene doc count. let me add comments for this.